web/src/client/software.js

```diff
@@ -217,78 +196,75 @@ class SoftwareBaseClient {
   /**
    * @param {string|undefined} address - D-Bus address; if it is undefined, it uses the system bus.
    */
-  constructor(address = undefined) {
-    this.client = new DBusClient(SOFTWARE_SERVICE, address);
-    this.product = new ProductManager(this.client);
-    this.proxies = {};
+  /**
+   * @param {import("./http").HTTPClient} client - HTTP client.
+   */
+  constructor(client) {
+    this.client = client;
   }
 
   /**
    * Asks the service to reload the repositories metadata
    *
    * @return {Promise<void>}
    */
-  async probe() {
-    const proxy = await this.client.proxy(SOFTWARE_IFACE);
-    return proxy.Probe();
+  probe() {
+    return this.client.post("/software/probe");
   }
 
   /**
    * Returns how much space installation takes on disk
    *
-   * @return {Promise<string>}
+   * @return {Promise<SoftwareProposal>}
    */
-  async getUsedSpace() {
-    const proxy = await this.client.proxy(SOFTWARE_IFACE);
-    return proxy.UsedDiskSpace();
+  getProposal() {
+    return this.client.get("/software/proposal");
   }
 
   /**
    * Returns available patterns
    *
-   * @param {boolean} filter - `true` = filter the patterns, `false` = all patterns
-   * @return {Promise<Array<string>>}
+   * @return {Promise<Pattern[]>}
    */
-  async patterns(filter) {
-    const proxy = await this.client.proxy(SOFTWARE_IFACE);
-    return proxy.ListPatterns(filter);
+  async getPatterns() {
+    /** @type Array<{ name: string, category: string, summary: string, description: string, order: string, icon: string }> */
+    const patterns = await this.client.get("/software/patterns");
+    return patterns.map((pattern) => ({
+      name: pattern.name,
+      category: pattern.category,
+      summary: pattern.summary,
+      description: pattern.description,
+      order: parseInt(pattern.order),
+      icon: pattern.icon,
+    }));
   }
 
   /**
-   * @typedef {Object.<string, number>} PatternSelection mapping "name" =>
-   * "who selected the pattern"
-   */
-
-  /**
-   * Returns selected patterns
-   *
-   * @return {Promise<PatternSelection>}
+   * @return {Promise<SoftwareConfig>}
    */
-  async selectedPatterns() {
-    const proxy = await this.client.proxy(SOFTWARE_IFACE);
-    return proxy.SelectedPatterns;
+  config() {
+    return this.client.get("/software/config");
   }
 
   /**
-   * Select a pattern to install
-   *
-   * @param {string} name - name of the pattern
+   * @param {Object.<string, boolean>} patterns - An object where the keys are the pattern names
```

Comment:
NP: This looks like something I have read above where the type didn't match with the description. Same type or just a copy/paste mistake?